torch/_dynamo/guards.py

```diff
@@ -2428,7 +2431,14 @@ def id_ref(self, obj):
                 # function, which will delete the callbacks as well. Therefore,
                 # we are using a finalizer which is kept alive.
                 self._weakrefs[id(obj)] = weakref.ref(obj)
-                weakref.finalize(obj, self.invalidate)
+
+                if obj.__str__ is object.__str__ and obj.__repr__ is object.__repr__:
+                    obj_str = str(obj)
+                else:
+                    obj_str = f"object with {id(obj)}"
```

Comment:
This will be `object with 1234` which doesn't sound right.

```suggestion
                obj_str = object.__str__(obj)
```